tendrl/common/utils/service.py

```diff
@@ -0,0 +1,50 @@
+from ansible_module_runner import AnsibleExecutableGenerationFailed
+from ansible_module_runner import AnsibleRunner
+import logging
+
+ANSIBLE_MODULE_PATH = "core/system/systemd.py"
+LOG = logging.getLogger(__name__)
+
+
+class Service(object):
+    def __init__(self, service_name, enabled=None):
+        self.attributes = {}
+        self.attributes["name"] = service_name
+
+        if enabled:
+            self.attributes["enabled"] = enabled
+
+    def __run_module(self, attr):
+        try:
+            runner = AnsibleRunner(
+                ANSIBLE_MODULE_PATH,
+                **attr
+            )
+            result, err = runner.run()
+        except AnsibleExecutableGenerationFailed as e:
+            LOG.error("Error switching the service: %s to %s state."
+                      " Error: %s" % (self.attributes["name"],
+                                      attr["state"],
+                                      str(e)))
+            return {}, e.message
+        return result, err
+
+    def start(self):
+        attr = self.attributes
+        attr.update({"state": "started"})
+        return self.__run_module(attr)
+
+    def stop(self):
+        attr = self.attributes
+        attr.update({"state": "stopped"})
+        return self.__run_module(attr)
+
+    def reload(self):
+        attr = self.attributes
+        attr.update({"state": "reloaded"})
+        return self.__run_module(attr)
+
+    def restart(self):
+        attr = self.attributes
+        attr.update({"state": "restarted"})
+        return self.__run_module(attr)
```

Comment:
The ansible module which we are using to manage the service has only 4 types of operations namely start, stop, restart and reload(http://docs.ansible.com/ansible/service_module.html refer state under options). So its not possible with this approach.
I feel when we monitor systemd dbus signals we will be able to obtain the service states